### ISSUE TITLE
Internal Server Error when creating a project with a Git-upstream that already contains a project

### ISSUE THREAD

**Issue by johannes-b** [keptn/keptn#3816]
**Environment**

* Keptn Version (`keptn version`): 0.8.1

**Affected Component**

* [ ] Docs
* [ ] CLI
* [ ] Kubernetes Integration
* [ ] Openshift Integration
* [ ] Helm
* [ ] Istio
* [ ] Bridge
* [ ] Approval
* [ ] Datastore
* [ ] REST API
* [x] shipyard-controller
* [ ] jmeter

**Describe the bug**

* I have already a GitHub repo that contains a project. 
* When creating a new Keptn project referencing this GitHub upstream repo, the shipyard-controller runs in an Internal Server Error. `
```
keptn create project sockshop3 --shipyard=./shipyard.yaml --git-user=keptn-nightly-user --git-token=***** --git-remote-url=https://github.com/my-keptn-tests/xyz.git
Starting to create project
Error: Create project was unsuccessful.
Received unexpected response: 500 500 Internal Server Error
```

Logs of shipyard-controller:
```
2021-04-19T15:35:56.300794539Z {"timestamp":"2021-04-19T15:35:56.300537627Z","logLevel":"INFO","message":"Created project in configuration service: sockshop3"}
2021-04-19T15:35:57.190127767Z 
2021-04-19T15:35:57.190184216Z 
2021-04-19T15:35:57.190191673Z 2021/04/19 15:35:57 [Recovery] 2021/04/19 - 15:35:57 panic recovered:
2021-04-19T15:35:57.190197358Z runtime error: invalid memory address or nil pointer dereference
2021-04-19T15:35:57.190201844Z /usr/local/go/src/runtime/panic.go:199 (0x44538b)
2021-04-19T15:35:57.190206519Z /usr/local/go/src/runtime/signal_unix.go:394 (0x4451c8)
2021-04-19T15:35:57.190210819Z /go/src/github.com/keptn/keptn/shipyard-controller/handler/projecthandler.go:159 (0x16b73c5)
2021-04-19T15:35:57.190215393Z /go/pkg/mod/github.com/gin-gonic/gin@v1.6.3/context.go:161 (0x98cdba)
2021-04-19T15:35:57.190220021Z /go/pkg/mod/github.com/gin-gonic/gin@v1.6.3/recovery.go:83 (0x9a0be3)
2021-04-19T15:35:57.190224632Z /go/pkg/mod/github.com/gin-gonic/gin@v1.6.3/context.go:161 (0x98cdba)
2021-04-19T15:35:57.190229276Z /go/pkg/mod/github.com/gin-gonic/gin@v1.6.3/logger.go:241 (0x99fd10)
2021-04-19T15:35:57.190233479Z /go/pkg/mod/github.com/gin-gonic/gin@v1.6.3/context.go:161 (0x98cdba)
2021-04-19T15:35:57.190237962Z /go/pkg/mod/github.com/gin-gonic/gin@v1.6.3/gin.go:409 (0x99702c)
2021-04-19T15:35:57.190254170Z /go/pkg/mod/github.com/gin-gonic/gin@v1.6.3/gin.go:367 (0x99672d)
2021-04-19T15:35:57.190260979Z /usr/local/go/src/net/http/server.go:2802 (0x6e9ab3)
2021-04-19T15:35:57.190266328Z /usr/local/go/src/net/http/server.go:1890 (0x6e5354)
2021-04-19T15:35:57.190270039Z /usr/local/go/src/runtime/asm_amd64.s:1357 (0x45d980)
2021-04-19T15:35:57.190274235Z 
```

I assume, the problem is because the `rollback` is nil what can happen here `err, rollback := ph.ProjectManager.Create(createProjectParams)` - please see: https://github.com/keptn/keptn/blob/master/shipyard-controller/handler/projecthandler.go#L159

**Expected behaviour:**
- [ ] As a user I get a notification that creating a project is not possible since *to-be-clarified

**Comment by warber** [comment]
@johannes-b I think this issue might already be fixed in current master: https://github.com/keptn/keptn/pull/3774

**Comment by johannes-b** [comment]
@warber thanks for the hint. Let's test it out on 0.8.2.

**Comment by johannes-b** [comment]
@warber correct, this issue is already fixed by: `keptn/shipyard-controller:0.8.2-dev-PR-3774.202104140833`
I closed it.
